skglm/utils/prox_funcs.py

```diff
@@ -60,15 +60,26 @@ def value_MCP(w, alpha, gamma):
 
 
 @njit
-def prox_MCP(value, stepsize, alpha, gamma):
-    """Compute the proximal operator of stepsize * MCP penalty."""
+def value_weighted_MCP(w, alpha, gamma, weights):
+    """Compute the value of the weighted MCP."""
+    s0 = np.abs(w) < gamma * alpha
+    value = np.zeros_like(w)
+    value[~s0] = weights[~s0] * gamma * alpha ** 2 / 2.
+    value[s0] = alpha * np.abs(w[s0]) - w[s0]**2 / (2 * gamma)
+    value[s0] *= weights[s0]
+    return np.sum(value)
+
+
+@njit
+def prox_MCP(value, stepsize, alpha, gamma, positive=False, weight=1.):
+    """Compute the proximal operator of stepsize * weight MCP penalty."""
     tau = alpha * stepsize
```

Comment:
prox of weight * MCP(gamma, alpha) is not prox of MCP(gamma, alpha * weight) so I don't get why only tau is multiplied by weight. Shouldn't gamma be divided too?

Reply:
The prox formula is correct as when ``tau`` and ``gamma`` are multiplied the ``weight`` cancel out.

However, I do agree with you, we better write
```python
tau = alpha * stepsize * weight
gamma = gamma / stepsize / weight
```
and remove the hardcoded `weight` in the remaining function body.

Reply:
+1 in order to handle 0 weights and avoid numerical issues, I do not call MCP(gamma / weight, alpha * weight) . Instead, I replaced "by hand" `alpha` by `alpha * weight` and `gamma` by `gamma / weight`.
This way, the weight cancels in `g * tau` here https://github.com/QB3/skglm/blob/461065eb7a1b96874629ade362beb6350be09b7a/skglm/utils/prox_funcs.py#L80